# Genome Assembly Bioininformatics Plan

You will be given a verbal introduction to the week's work, and we will describe the analyses you will need to complete and the associated assessed competency work.

We will start with the notebook `1.intro_to_unix_command_line.ipynb`. Open and read, trying the examples for yourself as you go through. This will teach you key command line skills you will need for data analysis.

Next we will use your new skills to examine the sequence data that you generated last term in notebook `2.sequence_QC.ipynb`. You will describe the sequence data, and quality control it retaining only the highest quality reads for further analysis.

Following this you are next asked to taxonomically identify the sequences using `3.taxonomic_ID.ipynb`. Waht species have you actually sequenced E. coli? How much contamination is there in your data?

You will then Assemble your bacterial genomes, and analyse the genomes you have assembled using `4.assembly.ipynb`. You will describe how many contigs are produced, how many reads were assembled, and several other key assembly statistics you will learn. This will be an important part of you Results section of your final submission.

At the end of this week you will have produced an assembled E. coli genome with a basic analyis of its completeness and quality. You will be able to say that you have "good basic data analysis and bioinformatics skills" on your cv, and be able to list an impressive set of specific bioinformatics analyses that you have done.
